rs/bitcoin/kyt/src/state.rs

```diff
@@ -122,3 +127,82 @@ impl Drop for FetchGuard {
         }
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct Config {
+    pub network: Network,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            network: Network::Bitcoin,
+        }
+    }
+}
+
+impl From<InitArg> for Config {
+    fn from(arg: InitArg) -> Self {
+        let network = match arg.network {
+            BtcNetwork::Mainnet => Network::Bitcoin,
+            BtcNetwork::Testnet => Network::Testnet,
+            BtcNetwork::Regtest => Network::Regtest,
+        };
+        Self { network }
+    }
+}
+
+impl From<&Config> for InitArg {
+    fn from(arg: &Config) -> Self {
+        let network = match arg.network {
+            Network::Bitcoin => BtcNetwork::Mainnet,
+            Network::Testnet => BtcNetwork::Testnet,
+            Network::Regtest => BtcNetwork::Regtest,
+            _ => panic!("unsupported network type: Signet"),
```

Comment:
If it is exhaustive, shouldn't it say:

```
Network::Signet => panic!("unsupported network type: Signet"),
```

Reply:
See the comment below regarding the implementation of `Storable` for `Config` but I don't think we should have this `From<&Config> for InitArg` conversion at all.